var/spack/repos/builtin/packages/ratel/package.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Ratel(MakefilePackage, CudaPackage, ROCmPackage):
+    """Extensible, performance-portable solid mechanics with libCEED and PETSc"""
+
+    homepage = "https://ratel.micromorph.org"
+    git = "https://gitlab.com/micromorph/ratel.git"
+
+    maintainers = ['jedbrown', 'jeremylt']
+
+    # development version
+    version('develop', branch='main')
+    depends_on('libceed@develop', when='@develop')
+    depends_on('petsc@main', when='@develop')
+
+    # version 0.1.1
+    version('0.1.1', tag='v0.1.1')
+    depends_on('libceed@0.10.1', when='@0.1.1')
+    depends_on('petsc@3.17.0', when='@0.1.1')
```

Comment:
We almost never use pinned versions in Spack, can you convert to a suitable range?

Reply:
There exists only one version that works on that release. The development version of PETSc is already not compatible

Reply:
> There exists only one version that works on that release. 

Fair, but I hope that at least new patches coming out in the future will be compatible. What is currently coded is that `ratel@0.1.1` works with `petsc@3.17.0` and not with `petsc@3.17.X` with X > 0. Same applies to `libceed`.

Reply:
@jeremylt This can be a range because petsc-3.17.x will continue to work.

@alalazo Is there an idiom for semver ranges or is it just dummy large values?
```suggestion
    version('0.1.1', tag='v0.1.1')
    depends_on('libceed@0.10.1:0.10.99', when='@0.1.1')
    depends_on('petsc@3.17.0:3.17.99', when='@0.1.1')
```

Reply:
> Is there an idiom for semver ranges

You can use:
```python
depends_on('petsc@3.17:3.17')
```